sdk/tables/azure-data-tables/azure/data/tables/_deserialize.py

```diff
@@ -174,3 +211,189 @@ def _extract_etag(response):
         return response.headers.get('etag')
 
     return None
+
+
+def normalize_headers(headers):
+    normalized = {}
+    for key, value in headers.items():
+        if key.startswith('x-ms-'):
+            key = key[5:]
+        normalized[key.lower().replace('-', '_')] = get_enum_value(value)
+    return normalized
+
+
+def return_headers_and_deserialized(response, deserialized, response_headers):  # pylint: disable=unused-argument
+    return normalize_headers(response_headers), deserialized
+
+
+def return_context_and_deserialized(response, deserialized, response_headers):  # pylint: disable=unused-argument
+    return response.http_response.location_mode, deserialized, response_headers
+
+
+def process_table_error(storage_error):
```

Comment:
just a thought, correct me if wrong: would it be better to have this in `_error.py`?

Reply:
Doing this will also create the circular import that I mentioned above